test/system/800-config.bats

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env bats   -*- bats -*-
+#
+# Test specific configuration options and overrides
+#
+
+load helpers
+
+@test "podman CONTAINERS_CONF - CONTAINERS_CONF in conmon {
+    skip_if_remote "can't check conmon environment over remote"
+
+    # Make an innocuous containers.conf in a non-standard location
+    export CONTAINERS_CONF="$PODMAN_TMPDIR/containers.conf"
+    cat >$CONTAINERS_CONF <<EOF
+[engine]
+runtime="$OCI_RUNTIME"
+EOF
+    run_podman run -d $IMAGE sleep infinity
+    cid="$output"
+
+    run_podman inspect "$cid" --format "{{ .State.ConmonPid }}"
+    conmon="$output"
+
+    output="$(tr '\0' '\n' < /proc/$conmon/environ | grep '^CONTAINERS_CONF=')"
+    is "$output" "CONTAINERS_CONF=$CONTAINERS_CONF"
+
+    # Clean up
+    # Weirdly, podman stop / SIGTERM doesn't seem to work here
+    run_podman kill "$cid"
+    run_podman rm "$cid"
+}
+
+@test "podman CONTAINERS_CONF - override runtime name" {
+    # Get the path of the normal runtime
+    run_podman info --format "{{ .Host.OCIRuntime.Path }}"
+    ocipath="$output"
+
+    export CONTAINERS_CONF="$PODMAN_TMPDIR/nonstandard_runtime_name.conf"
+    cat > $CONTAINERS_CONF <<EOF
+[engine]
+runtime = "nonstandard_runtime_name"
+[engine.runtimes]
+nonstandard_runtime_name = ["$ocipath"]
+EOF
+
+    run_podman run -d --rm $IMAGE true
+    cid="$output"
+
+    run_podman 1 rm "$cid"
+    # Weirdly, the error message is different for root vs. rootless
+    is "$output" \
+        "Error:.*\(no container\|no such container\)"
```

Comment:
You've run the container with `--rm`. This is a NOP.

Reply:
It *should* be a no-op, but precisely the bug I'm addressing here means it isn't without this fix.